administration/src/bp-modules/applications/utils/exportApplicationToCsv.ts

```diff
@@ -0,0 +1,63 @@
+import { stringify } from 'csv-stringify/browser/esm/sync'
+
+import i18next from '../../../i18n'
+import { ProjectConfig } from '../../../project-configs/getProjectConfig'
+import {
+  AddressApplicationData,
+  ApplicationDataIncompleteError,
+  CardTypeApplicationData,
+  CreationDateApplicationData,
+  PersonalApplicationData,
+  getAddressApplicationData,
+  getCardTypeApplicationData,
+  getPersonalApplicationData,
+} from '../../../util/applicationDataHelper'
+import downloadDataUri from '../../../util/downloadDataUri'
+import formatDateWithTimezone from '../../../util/formatDate'
+import type { JsonField } from '../JsonFieldView'
+import { CSV_MIME_TYPE_UTF8 } from '../constants'
+import { GetApplicationsType } from '../types'
+
+export type CSVApplicationData = PersonalApplicationData &
+  AddressApplicationData &
+  CardTypeApplicationData &
+  CreationDateApplicationData
+
+export class ApplicationToCsvError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'ApplicationToCsvError'
+  }
+}
+
+export const exportApplicationToCsv = (application: GetApplicationsType, config: ProjectConfig): void => {
+  try {
+    if (!config.applicationFeature?.csvExport) {
+      throw new ApplicationToCsvError('This project does not support application CSV export.')
+    }
+    const json: JsonField<'Array'> = JSON.parse(application.jsonValue)
+
+    const csvData: CSVApplicationData = {
+      ...getPersonalApplicationData(json),
+      ...getAddressApplicationData(json),
+      ...getCardTypeApplicationData(json),
+      ...{ creationDate: formatDateWithTimezone(application.createdDate, config.timezone) },
+    }
+
+    const csvHeader = Object.keys(csvData).map(key => i18next.t(`application:${key}`))
+    const csvContent = Object.values(csvData)
+    const csv = stringify([csvHeader]) + stringify([csvContent])
+
+    const blob = new Blob([csv], { type: CSV_MIME_TYPE_UTF8 })
+    const { surname, forenames, creationDate } = csvData
+    downloadDataUri(blob, `${surname}_${forenames}_${creationDate}.csv`)
```

Comment:
🤔 I was trying to add a special char to surname like ? or / and that got handle by the OS the file name adjusted and the char became `__` not sure if other operating systems could do the same.
Do we need to sanitize or no need for that?

Reply:
Hm we also use this naming format for PDFs somewhere I think. If there is no error I don't think this is super important